server/segment_rpc_test.go

```diff
@@ -1233,181 +1265,56 @@ func TestSubmitSegment_Success(t *testing.T) {
 	balance.AssertCalled(t, "Credit", ratMatcher(change))
 }
 
-func TestSubmitSegment_UpdateOrchestratorInfo(t *testing.T) {
-	require := require.New(t)
-
-	params := pm.TicketParams{
-		Recipient:         ethcommon.Address{},
-		FaceValue:         big.NewInt(100),
-		WinProb:           big.NewInt(100),
-		RecipientRandHash: pm.RandHash(),
-		Seed:              big.NewInt(100),
-	}
+func TestUpdateOrchestratorInfo(t *testing.T) {
+	assert := assert.New(t)
 
-	tr := &net.TranscodeResult{
-		Result: &net.TranscodeResult_Data{
-			Data: &net.TranscodeData{
-				Segments: []*net.TranscodedSegmentData{
-					&net.TranscodedSegmentData{Url: "foo", Pixels: 10},
-				},
-				Sig: []byte("bar"),
-			},
-		},
-		Info: &net.OrchestratorInfo{
-			Transcoder: "http://google.com",
-			PriceInfo: &net.PriceInfo{
-				PricePerUnit:  1,
-				PixelsPerUnit: 1,
-			},
-			TicketParams: &net.TicketParams{
-				Recipient:         params.Recipient.Bytes(),
-				FaceValue:         params.FaceValue.Bytes(),
-				WinProb:           params.WinProb.Bytes(),
-				RecipientRandHash: params.RecipientRandHash.Bytes(),
-				Seed:              params.Seed.Bytes(),
+	sess := &BroadcastSession{PMSessionID: "foo"}
+	info := &net.OrchestratorInfo{
+		Storage: []*net.OSInfo{
+			&net.OSInfo{
+				StorageType: 1,
+				S3Info:      &net.S3OSInfo{Host: "http://apple.com"},
 			},
 		},
 	}
-	buf, err := proto.Marshal(tr)
-	require.Nil(err)
 
-	ts, mux := stubTLSServer()
-	defer ts.Close()
-	mux.HandleFunc("/segment", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		w.Write(buf)
-	})
-
-	// Test with no pm.Sender in BroadcastSession
-
-	newSess := func() *BroadcastSession {
-		return &BroadcastSession{
-			Broadcaster: stubBroadcaster2(),
-			ManifestID:  core.RandomManifestID(),
-			OrchestratorInfo: &net.OrchestratorInfo{
-				Transcoder: ts.URL,
-				PriceInfo: &net.PriceInfo{
-					PricePerUnit:  2,
-					PixelsPerUnit: 1,
-				},
-			},
-		}
-	}
-
-	s := newSess()
-
-	assert := assert.New(t)
-
-	_, err = SubmitSegment(s, &stream.HLSSegment{Data: []byte("dummy")}, 0)
-
-	assert.Nil(err)
-	assert.Equal("http://google.com", s.OrchestratorInfo.Transcoder)
-	assert.Equal(tr.Info.TicketParams.Recipient, s.OrchestratorInfo.TicketParams.Recipient)
-	assert.Equal(tr.Info.TicketParams.FaceValue, s.OrchestratorInfo.TicketParams.FaceValue)
-	assert.Equal(tr.Info.TicketParams.WinProb, s.OrchestratorInfo.TicketParams.WinProb)
-	assert.Equal(tr.Info.TicketParams.RecipientRandHash, s.OrchestratorInfo.TicketParams.RecipientRandHash)
-	assert.Equal(tr.Info.TicketParams.Seed, s.OrchestratorInfo.TicketParams.Seed)
```

Comment:
The `TicketParams` assertions were omitted in this PR in favor of just checking that the `OrchestratorInfo` field of a session was updated (`assert.Equal()` conveniently compares referenced values when pointers are passed in) - see [this test](https://github.com/livepeer/go-livepeer/pull/1254/files#diff-f986f75ea9a0ee09429050246648e0b7R640).

A few of the removed tests were previously included to make sure that the `OrchestratorInfo` field of a session was not updated until the very end of `SubmitSegment()`. This [test](https://github.com/livepeer/go-livepeer/pull/1254/files#diff-6a2971a2e6142892f2da0c440d15a274L1340) is one such example - we checked that the local balance update performed in `SubmitSegment()` is calculated based on the price *before* receiving a new price in an updated `OrchestratorInfo` message. Since we no longer mutate a session within `SubmitSegment()`, we don't need these types of tests anymore.